src/riscv-integration.adoc

```diff
@@ -791,6 +805,26 @@ CHERI violations have the following order in priority:
 . Invalid address violation
 . Bounds violation (_Lowest_)
 
+[#mtval2,reftext="mtval2"]
+==== Machine Trap Value Register 2 (mtval2)
+
+The <<mtval2>> register is an MXLEN-bit read-write register, which is added as part of the
+Hypervisor extension. {cheri_base_ext_name} also requires the implementation of this CSR.
+
+When a CHERI fault is taken on any data memory access into into M-mode, <<mtval2>> is written
+with the MXLEN-bit effective address which caused the fault. This follows the existing rules for reporting
+load/store addresses to <<mtval>> from [riscv-priv-spec].
+//This reported address can be used together with the CAP_IDX field of <<mtval>> to diagnose the
+//authorising capability which caused the exception.
```

Comment:
Could we remove the comment before merging?

Reply:
feel free